roles_revamp.md

```diff
@@ -0,0 +1,37 @@
+# Proposal: Revamp Roles
+
+*Author*: Brian Ccoai <@bcoca>
+
+*Date*: 04/01/2016
+
+- Status: New
+- Proposal type: core design
+- Targeted Release: 2.2
+- PR for Comments: 
+- Estimated time to implement: 4 weeks
+
+
+## Motivation
+Roles are a good way to share and reuse resources, but they are a bit limited in application.
+This proposal aims at expanding the usefulness and reusability of roles.
+
+### Problems
+Roles lack flexibility in execution and inclusion.
+
+## Solution proposal
+- Add exec_main flag to roles (default True to keep current behaviour), when false a role will not execute tasks/main.yml in `roles:` section of a play.
```

Comment:
This is only needed if a role has a `tasks/main.yml` and the user does not want it to run correct? Essentially packaging task files _not_ named main.yml would have the same effect.

Also, I'm not sure `exec_main` semantically. Living in core I can see where you think in terms of main, but most users would think in terms of tasks. Plus there are other mains in a role -- handlers/main.yml, vars/main.yml etc. I'd suggest `exec_tasks` as a boolean with True being the default and something end users could grasp more easily.